invenio/ext/login/__init__.py

```diff
@@ -151,11 +156,11 @@ def do_login_first(error=401):
         if not urllib.unquote(secure_url).startswith(request.base_url):
             return redirect(secure_url)
         if current_user.is_guest:
-            flash(g._("Please sign in to continue."), 'info')
+            if not session.get('_flashes'):
+                flash(g._("Please sign in to continue."), 'info')
             from invenio.modules.accounts.views.accounts import login
-            return login(referer=request.url), 401
+            return login(referer=request.url)
```

Comment:
found it here https://github.com/inveniosoftware/invenio/pull/2873/files#diff-d0cf81b2920b7bd09d191730c0433305R101 :+1: